modules/integration/src/test/java/io/ballerina/messaging/broker/integration/standalone/jms/QueueLocalTransactionRollbackTest.java

```diff
@@ -37,55 +38,19 @@
 import javax.naming.InitialContext;
 import javax.naming.NamingException;
 
-
 /**
- * Class for testing local transactions commit and rollback operations in JMS
+ * Test cases written to verify queue consumer and producer behavior in local transaction rollback operation
  */
-public class JmsLocalTransactionTest {
-
-    @Parameters({"broker-port"})
-    @Test
-    public void testConsumerProducerCommitTransaction(String port) throws NamingException, JMSException {
-        String queueName = "testCommitConsumerAndPublisherTransaction";
-        InitialContext initialContextForQueue = ClientHelper
-                .getInitialContextBuilder("admin", "admin", "localhost", port)
-                .withQueue(queueName)
-                .build();
-
-        ConnectionFactory connectionFactory
-                = (ConnectionFactory) initialContextForQueue.lookup(ClientHelper.CONNECTION_FACTORY);
-        Connection connection = connectionFactory.createConnection();
-        connection.start();
-
-        // publish 100 messages
-        Session producerSession = connection.createSession(true, Session.SESSION_TRANSACTED);
-        Queue queue = producerSession.createQueue(queueName);
-        MessageProducer producer = producerSession.createProducer(queue);
+public class QueueLocalTransactionRollbackTest {
 
-        int numberOfMessages = 100;
-        for (int i = 0; i < numberOfMessages; i++) {
-            producer.send(producerSession.createTextMessage("Test message " + i));
-        }
-        //commit all sent messages
-        producerSession.commit();
-        producerSession.close();
-
-        // Consume published messages
-        Session subscriberSession = connection.createSession(true, Session.SESSION_TRANSACTED);
-        Destination subscriberDestination = (Destination) initialContextForQueue.lookup(queueName);
-        MessageConsumer consumer = subscriberSession.createConsumer(subscriberDestination);
-
-        for (int i = 0; i < numberOfMessages; i++) {
-            Message message = consumer.receive(5000);
-            Assert.assertNotNull(message, "Message #" + i + " was not received");
-        }
-        //commit all received messages
-        subscriberSession.commit();
-        Message message = consumer.receive(5000);
-        Assert.assertNull(message, "Messages should not receive after commit all");
+    @BeforeClass
+    public void setup() {
+        System.setProperty("STRICT_AMQP", "true");
+    }
 
-        subscriberSession.close();
-        connection.close();
+    @AfterClass
+    public void destroy() {
+        System.setProperty("STRICT_AMQP", "false");
```

Comment:
Isn't it better to call `System.clearProperty()` here?

Reply:
Fixed in 3ea6243